### PR TITLE
Fail test when deletion of test data folder failing.

### PR DESCRIPTION
**Pull request by MishaDemianenko**
Previous behavior was too permissive and as result,
several times real problems with file descriptors leaks were missed.

### CODE REVIEW

**Review by ragadeeshu — APPROVED** [review]

**Comment by MishaDemianenko** [comment]
@ragadeeshu i have made a couple of additional cleanups after your last review so would be good if you would make another round! Thx in advance!

File: community/kernel/src/test/java/org/neo4j/test/Unzip.java [hunk 1]
```diff
@@ -45,30 +45,31 @@ public static File unzip( Class<?> testClass, String resource, File targetDirect
 
         try
         {
-            ZipInputStream zipStream = new ZipInputStream( source );
-            ZipEntry entry = null;
-            byte[] scratch = new byte[8096];
-            while ( (entry = zipStream.getNextEntry()) != null )
+            try ( ZipInputStream zipStream = new ZipInputStream( source ) )
```
Comment (ragadeeshu) [inline community/kernel/src/test/java/org/neo4j/test/Unzip.java]: Double try

**Review by ragadeeshu — COMMENTED** [review]
@MishaDemianenko nice cleanups. Found a tiny thing :)

File: community/kernel/src/test/java/org/neo4j/test/Unzip.java (hunk 1 above)
Comment (MishaDemianenko) [reply community/kernel/src/test/java/org/neo4j/test/Unzip.java]: thx! cleaned up!

**Pull request merged**